sdk/core/Azure.Core.TestFramework/src/TestProxy.cs

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+using NUnit.Framework;
+
+namespace Azure.Core.TestFramework
+{
+    /// <summary>
+    /// Encapsulates a process running an instance of the Test Proxy as well as providing access to the Test Proxy administration actions
+    /// via the TestProxyRestClient.
+    /// <seealso href="https://github.com/Azure/azure-sdk-tools/tree/main/tools/test-proxy"/>
+    /// </summary>
+    public class TestProxy
+    {
+        private static bool HasDotNetExe(string dotnetDir) => dotnetDir != null && File.Exists(Path.Combine(dotnetDir, s_dotNetExeName));
+
+        private static readonly string s_dotNetExeName;
+
+        private static readonly string s_dotNetExe;
+
+        public const string DevCertIssuer = "CN=localhost";
+
+        public const string IpAddress = "127.0.0.1";
+
+        public int? ProxyPortHttp => _proxyPortHttp;
```

Comment:
Make these non-nullable and throw?

Reply:
Eh, I'd still need to check for 0 in that case.